geopandas/array.py

```diff
@@ -295,6 +295,18 @@ def __init__(self, data, crs=None):
 
     @property
     def data(self):
+        warnings.warn(
+            "Accessing the underlying geometries through the `.data` attribute is "
+            "deprecated and will be removed in GeoPandas 1.0. You can use "
+            "`np.asarray(..)` or the `to_numpy()` method instead.\n"
+            "Note that if you are using PyGEOS and using this attribute to get an "
+            "array of PyGEOS geometries, those other methods will always return an "
+            "array of Shapely geometries. Accessing the underlying PyGEOS geometries "
+            "directly is deprecated, and you should migrate to use Shapely >= 2.0 "
+            "instead.",
+            FutureWarning,
```

Comment:
We could also start with a DeprecationWarning (and change it to more visible FutureWarning in 0.14), so that users of other _libraries_ that do this don't yet see the warning (only for direct usage themselves)